node/pkg/watchers/sui/msg_verifier.go

```diff
@@ -0,0 +1,58 @@
+package sui
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"go.uber.org/zap"
+)
+
+func (e *Watcher) verify(
+	ctx context.Context,
+	msg *common.MessagePublication,
+	txDigest string,
+	logger *zap.Logger,
+) (common.MessagePublication, error) {
+
+	if msg == nil {
+		return common.MessagePublication{}, fmt.Errorf("MessagePublication is nil")
+	}
+
+	if msg.VerificationState() != common.NotVerified {
+		return common.MessagePublication{}, fmt.Errorf("MessagePublication already has a non-default verification state")
+	}
+
+	if e.suiTxVerifier == nil {
+		return common.MessagePublication{}, fmt.Errorf("transfer verifier is nil")
+	}
+
+	localMsg := msg
+
+	var verificationState common.VerificationState
+
+	// If the emitter address of the message does not match the token bridge emitter,
+	// mark the message's verification state as NotApplicable.
+	if localMsg.EmitterAddress.String() != e.suiTxVerifier.GetTokenBridgeEmitter() {
+		verificationState = common.NotApplicable
```

Comment:
This function should also return `NotApplicable` if the payload isn't a token transfer. (via `vaa.IsTransfer`). I see this is missing in the EVM side too, so we can make a follow-up PR to add it there.

Reply:
Good callout!